oshi-demo/src/main/java/oshi/demo/jmx/mbeans/BaseBoardMBean.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 The OSHI Project Contributors: https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.demo.jmx.mbeans;
+
+public interface BaseBoardMBean {
```

Comment:
I'm wondering if rather than manually writing these MBeans you can use Jackson's `ObjectMapper` to accomplish the same purpose.  Otherwise that's a ton of boilerplate code.

Reply:
ohh  got it, will work over it, to use Jacksons ObjectMapper

Reply:
Hi Already chnaged all the others comments,  the only one pending is this one.

Reply:
Sorry, busy today.  

No specific suggestion here.  I'm just thinking from the experience of the oshi-json artifact that maintaining a whole "mirror" of the API is a lot of work.  I was hoping ObjectMapper could directly communicate with JMX without coding up the beans.

Reply:
I see some problems with those approach, but here are some solutions. 

1 - we can convert the oshi objects to Json, and create  DynamicMBeaans, this mbean will accept an oshi  json representation and the Dynamic MBean  will just set up that mbean with the json provided, but the problem here is that we would need to always keep updating that instance of DynamicMBeans since it wont hold any actual instance 

2- we can deal with reflection, we can pass the actual object and execute methods by reflection and sets the fields of this DynamicMBean by iterating over each field of each object and that way we can call also the update method on each oshi object  to keep the obj updateded

3 or we can make an empty interface eg BaseboardMBean  and that interface will extend the Baseboard interface,  that way any implementation from windows , mac , linux could implement the BaseBoardMBean and refactor the code to just offering a method for retrive the the MBean representation of whatever object , this new method could be only available at the oshi package level so anyone can't use it

Reply:
Which option do you recommend?

OSHI already has interfaces that define what gets returned, which is essentially how Jackson converts them to its object.  The DynamicMBean approach sounds to me like a reasonable option, but you have a lot more experience here so I'd like to hear your thoughts.

Reply:
well, rethinking more about what you had proposed, Im thinking in using [ASM](https://asm.ow2.io/) with a maven steps in which we can grab all the interfaces on the corresponding classpath,  parse those clases into JSON with object mapper and later create the MBeans dinamycaally , so when the strategy for regsitering MBeans is executed, can have those MBenas  ref, will work on that

Reply:
beacuese sadly for JMX we need interface with the suffix MBena and the DynamicMBean would have been a good option but we would need to mantain some enums to specify the operation on each interface and  that would end up in the same problem

Reply:
Well at the end I went with the approach of DynamicMBean and reflection, so it resloves the overhead of creating repetitive  code. And for giving the users the capability to know what properties are in each MBena I offer an Interface for giving them the list of props they can retrive for each MBEan

Reply:
I'm completely in for this effort about creating these servers relying on OSHI and from my perspective that I worked in System Managment systems, it is really hard to find these Open Source efforts well mantained and documented. Actually we struggle a lot with [sblim-sfcb](http://sblim.sourceforge.net/wiki/index.php/Sfcb) beacuse either you need to pay for support at the different Linux distos or you need to have  c developers. and thks to you and all this comunity I came up with this awsome  library that helped the businnes at that time. And with Converting this into  complete server , the posibility are endlees, from either sending threshold event and more.   And Regarding this effort, for overcoming the need to construct these interfaces I just implemented and approach with reflection, that way we can have the DynamicMBeams always in sync with watever change is made to the oshi interfaces